### PR TITLE
fix: ACNA-1229 - failing unit test after changes on utils.getActionUrls

### PR DESCRIPTION
**Pull request by florind12**
## Description
The default package name will be included in the action name.

The idea behind the `legacy` flag is to support the deprecated URL key.

By default `getActionUrls` (`legacy` = false) will return the object with the new `pk/action` key as required, this alone will change the output of dist, and the output of `config.js` (breaking the compatibility with legacy keys)

When (`legacy` = true) the only extra change from the default output is adding the legacy key back to the config.js.

## :warning: Pre-requisite ⚠️ 

To be merged only after https://github.com/adobe/aio-lib-runtime/pull/76 is merged && released. / update package json. 
The unit tests will fail until we get the new aio-lib-runtime release.

Note:
Unit tests fail due to `aio-lib-runtime` pending changes. 

## Related Issue
https://github.com/adobe/aio-lib-runtime/pull/76
https://github.com/adobe/aio-lib-runtime/issues/64

## How Has This Been Tested?
Given aio generated app with the following packages. 

```
packages:
    tmp1:
      license: Apache-2.0
      actions:
        generic:
          function: actions/generic/index.js
          ...
    tmp2:
      license: Apache-2.0
      actions:
        another-generic:
          function: actions/another-generic/index.js
          ...
```
            
run `aio app run` || `aio app build` on an `dx/excshell/1` extension app. 
Expected: 
`/src/dx-excshell-1/web-src/src/config.json` should contain 
```
{
"generic":".../api/v1/web/tmp1/generic",
"tmp1/generic":".../api/v1/web/tmp1/generic",
"tmp2/another-generic":".../api/v1/web/tmp2/another-generic"
}
```
Where `generic` is the legacy key and the `tmp1/generic` and `tmp2/another-generic` are the new keys.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by sandeep-paliwal — APPROVED** [review]
LGTM
wait for the changes in runtime lib to get merged.

**Review by shazron — APPROVED** [review]

File: src/lib/run-dev.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ async function runDev (config, dataDir, options = {}, log = () => {}) {
         // note the condition: we still write backend urls EVEN if skipActions is set
         // the urls will always point to remotely deployed actions if skipActions is set
         log('injecting backend urls into frontend config')
-        urls = rtLibUtils.getActionUrls(devConfig, true, isLocal && !skipActions)
+        urls = rtLibUtils.getActionUrls(devConfig, true, isLocal && !skipActions, true)
```
Comment (shazron) [inline src/lib/run-dev.js:98]: we should discuss when we are turning "legacy = false"

Comment (florind12) [reply src/lib/run-dev.js:98]: The idea behind the legacy flag is to support the deprecated URL key.
By default `getActionUrls` will return the object with the new pk/action key as required, this alone will change the output of dist, and the output of `config.js`

So this flag will change the default output of the config.js, adding the support for the deprecated key.

**Comment by shazron** [comment]
The aio-lib-runtime changes have been merged, but not released yet.
@florind12 there is one unit-test failure now

**Pull request merged**